### PR TITLE
Bgen Changes 1

### PR DESCRIPTION
**Pull request by jigold**
- Write new index file to *.idx2
- Removed file_row_idx. Now take a list of variants to filter to.

Future Plans:
- Have import_bgen get reference_genome, contig_recoding, and skip_invalid_loci from index file instead of from user
- Change index_bgen and import_bgen interface with more flexibility on where to read/write index files from
- Filtering variants from a Table -- User specifies expression from an existing table or a list of variants to filter from.

### CODE REVIEW

**Comment by danking** [comment]
I don't really understand the failure. Seems like the stack trace is missing?

**Comment by danking** [comment]
@jigold I also made bien-changes protected so that we have to go through a PR process. Is that OK?

**Comment by jigold** [comment]
Sure. Sounds good!

File: python/hail/expr/expressions/base_expression.py [hunk 1]
```diff
@@ -581,6 +581,14 @@ def __ne__(self, other):
                             f"of type '{self.dtype}' and '{other.dtype}'")
         return left._bin_op("!=", right, tbool)
 
+    def _to_table_wout_keys(self, name, resort=False):
```
[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/expr/expressions/base_expression.py]: I need a way to convert an arbitrary expression to a table such as `hl.struct(locus=mt.locus2, alleles=mt.alleles2)` where `locus2` and `alleles2` aren't row keys without including any other row keys in the resulting table. But the result must be sorted by the new variant expression.

[the remark this answers is not in the document]
Comment (patrick-schultz) [reply python/hail/expr/expressions/base_expression.py]: Ah, sorry, I misunderstood.

One small optimization you could make: in the `resort` case, the `key_by` is shuffling the old keys, which you then drop. You could avoid that by
```
t = self._to_table(name)
t = t.key_by().select(name)
if resort:
    t = t.key_by(name)
return t
```

It's annoying that we have to copy the old keys into `t` in the first place. @jbloom22 just ran into this problem too. Something like an `include_keys` option on `_to_table` would let us optimize cases like this where you don't care about the keys. Anyways, not suggesting you should do that in this PR, just noticed the same issue coming up.

**Comment by jigold** [comment]
Closing in favor of #4291

**Pull request closed without merging**